### PR TITLE
migrate the docs

### PR DESCRIPTION
**Pull request by stan-dot**
- [ ] squash the commits that change the contents - (all except the first one)

### CODE REVIEW

**Comment by stan-dot** [comment]
> /home/runner/work/blueapi/blueapi/docs/reference.md:5: WARNING: toctree contains reference to document 'reference/cli' that doesn't have a title: no link will be generated
one small error with toctree extension - might be rst specific, need equivalent for md

**Review by callumforrester — CHANGES_REQUESTED** [review]
Git appears to see these as deleting old files and creating new ones, rather than renaming. Can we somehow preserve history?

**Comment by stan-dot** [comment]
we cannot preserve that, renaming the file extension makes the change.

**Comment by DiamondJoseph** [comment]
> we cannot preserve that, renaming the file extension makes the change.

Can you try renaming the files, committing, then replacing the contents, committing again?

**Comment by stan-dot** [comment]
why do we care about the exact history? this app has only been run on 1 live beamline.

I'll check the history thing on Tuesday if time allows

**Comment by callumforrester** [comment]
This needs to be two commits:
1. Changes the file names
2. Changes the file contents
It then needs to be merged _without_ squashing

**Comment by stan-dot** [comment]
that here is quite odd

**Comment by stan-dot** [comment]
todo: need to remove the sphinx dependency and the CI jobs: `ERROR: InvocationError for command /opt/hostedtoolcache/Python/3.11.9/x64/bin/sphinx-build -EW --keep-going -T docs build/html (exited with code 1)`

@callumforrester will the new markdown docs just livein the repo or do we build them and deploy to some place like github pages?

**Review by callumforrester — APPROVED** [review]

File: docs/reference/rest-spec.md
Comment (DiamondJoseph) [inline docs/reference/rest-spec.md:1]: Don't these files require changes?

File: docs/reference/messaging-spec.md
Comment (DiamondJoseph) [inline docs/reference/messaging-spec.md:1]: Don't these files require changes?

File: docs/reference/cli.md
Comment (DiamondJoseph) [inline docs/reference/cli.md:1]: Don't these files require changes?

**Review by DiamondJoseph — CHANGES_REQUESTED** [review]

File: docs/reference/rest-spec.md
Comment (stan-dot) [reply docs/reference/rest-spec.md:1]: ok will review those all

File: docs/reference/messaging-spec.md
Comment (stan-dot) [reply docs/reference/messaging-spec.md:1]: ok will review those all

File: docs/reference/cli.md
Comment (stan-dot) [reply docs/reference/cli.md:1]: ok will review those all

**Comment by stan-dot** [comment]
what is this??? 
image

File: docs/reference/messaging-spec.md
Comment (stan-dot) [reply docs/reference/messaging-spec.md:1]: no idea what is this about, :show-nested:, it couldn't be less informative

File: docs/reference/rest-spec.md
Comment (stan-dot) [reply docs/reference/rest-spec.md:1]: this done ok

File: docs/reference/cli.md
Comment (stan-dot) [reply docs/reference/cli.md:1]: no idea what is this about, :show-nested:, it couldn't be less informative

File: docs/reference/messaging-spec.md
Comment (stan-dot) [reply docs/reference/messaging-spec.md:1]: done

**Comment by DiamondJoseph** [comment]
> what is this??? 

It appears to be embedding the rst inside the md rather than doing the equivalent md formatting :D

**Comment by stan-dot** [comment]
> > what is this???
> 
> It appears to be embedding the rst inside the md rather than doing the equivalent md formatting :D

I know that , it's an artifact from a fully rst times. But what sorcery did it do behind the scenes to dynamically import version like that? Do we want to support this in md?

**Comment by DiamondJoseph** [comment]
It's part of the configured tooling.
https://github.com/DiamondLightSource/blueapi/blob/main/pyproject.toml#L72

I think it is correct for it to be there (it is available to call `blueapi version` after pip installing)

**Comment by stan-dot** [comment]
I don't think that markdown calls this rightly, this is redundant information that could be deleted from `api.md`

@DiamondJoseph

**Comment by stan-dot** [comment]
f

**Comment by stan-dot** [comment]
following [this comment](https://github.com/sphinx-doc/sphinx/issues/11739#issuecomment-1793687223)  to get it to work

**Review by callumforrester — APPROVED** [review]
LGTM, thanks!

**Pull request merged**